docs/assets/dcr_tmux_simnet_setup.sh

```diff
@@ -0,0 +1,275 @@
+#
+# Both wallet1 and wallet2 use the same seed, however, wallet1 is configured to
+# automatically buy tickets and vote, while wallet2 is only configured vote.
+#
+# The primary dcrd node (dcrd1) is configured as the primary mining node.
+#
+# Network layout:
+# dcrd1 (p2p: localhost:19555) <-> dcrd2 (p2p: localhost:19565)
+#
+# RPC layout:
+# dcrd1 (JSON-RPC: localhost:19556)
+#     ^--- wallet1 (JSON-RPC: locahost:19557, gRPC: localhost:19558)
+# dcrd2 (JSON-RPC: localhost:19566)
+#     ^--- wallet2 (JSON-RPC: locahost:19567, gRPC: None)
+
+set -e
+
+SESSION="dcrd-simnet-nodes"
+NODES_ROOT=~/dcrdsimnetnodes
+RPCUSER="USER"
+RPCPASS="PASS"
+WALLET_SEED="b280922d2cffda44648346412c5ec97f429938105003730414f10b01e1402eac"
+WALLET_MINING_ADDR="SspUvSyDGSzvPz2NfdZ5LW15uq6rmuGZyhL" # NOTE: This must be changed if the seed is changed.
+WALLET_XFER_ADDR="SsonWQK6xkLSYm7VttCddHWkWWhETFMdR4Y" # same as above
+
+if [ -d "${NODES_ROOT}" ] ; then
+  rm -R "${NODES_ROOT}"
+fi
+
+PRIMARY_DCRD_NAME=dcrd1
+SECONDARY_DCRD_NAME=dcrd2
+PRIMARY_WALLET_NAME=wallet1
+SECONDARY_WALLET_NAME=wallet2
+mkdir -p "${NODES_ROOT}/${PRIMARY_DCRD_NAME}"
+mkdir -p "${NODES_ROOT}/${SECONDARY_DCRD_NAME}"
+mkdir -p "${NODES_ROOT}/${PRIMARY_WALLET_NAME}"
+mkdir -p "${NODES_ROOT}/${SECONDARY_WALLET_NAME}"
+
+cat > "${NODES_ROOT}/dcrd.conf" <<EOF
+rpcuser=${RPCUSER}
+rpcpass=${RPCPASS}
+simnet=1
+logdir=./log
+datadir=./data
+debuglevel=TXMP=debug,MINR=debug
+txindex=1
+EOF
+
+cat > "${NODES_ROOT}/dcrctl.conf" <<EOF
+rpcuser=${RPCUSER}
+rpcpass=${RPCPASS}
+simnet=1
+EOF
+
+cat > "${NODES_ROOT}/wallet.conf" <<EOF
+username=${RPCUSER}
+password=${RPCPASS}
+simnet=1
+logdir=./log
+appdata=./data
+pass=123
+enablevoting=1
+EOF
+
+cd ${NODES_ROOT} && tmux -2 new-session -d -s $SESSION
+
+################################################################################
+# Setup the primary dcrd node
+################################################################################
+
+PRIMARY_DCRD_P2P=127.0.0.1:19555
+PRIMARY_DCRD_RPC=127.0.0.1:19556
+tmux rename-window -t $SESSION:0 "${PRIMARY_DCRD_NAME}"
+tmux split-window -v
+tmux select-pane -t 0
+tmux send-keys "cd ${NODES_ROOT}/${PRIMARY_DCRD_NAME}" C-m
+tmux send-keys "dcrd -C ../dcrd.conf --listen ${PRIMARY_DCRD_P2P} --miningaddr=${WALLET_MINING_ADDR}" C-m
+tmux resize-pane -D 15
+tmux select-pane -t 1
+tmux send-keys "cd ${NODES_ROOT}/${PRIMARY_DCRD_NAME}" C-m
+
+cat > "${NODES_ROOT}/${PRIMARY_DCRD_NAME}/ctl" <<EOF
+#!/usr/bin/env bash
+dcrctl -C ../dcrctl.conf \$*
+EOF
+chmod +x "${NODES_ROOT}/${PRIMARY_DCRD_NAME}/ctl"
+
+# Script to mine a specified number of blocks with a delay in between them
+# Defaults to 1 block
+cat > "${NODES_ROOT}/${PRIMARY_DCRD_NAME}/mine" <<EOF
+#!/usr/bin/env bash
+NUM=1
+case \$1 in
+    ''|*[!0-9]*)  ;;
+    *) NUM=\$1 ;;
+esac
+
+for i in \$(seq \$NUM) ; do
+  ./ctl generate 1
+  sleep 1
+done
+EOF
+chmod +x "${NODES_ROOT}/${PRIMARY_DCRD_NAME}/mine"
+sleep 3
```

Comment:
The sleeps in this script aren't large so it's usually not an issue but on my dcrlnd tmux scripts (where there are larger waits) I've taken to using the following function instead of simply a sleep so I have a better idea of where the execution is (it looks better when doing a presentation as well :P)
```bash
#!/bin/bash

function countdown {
        secs=$1
        msg=$2
        while [ $secs -gt 0  ]; do
                echo -ne "$secs $msg\033[0K\r"
                sleep 1
                : $((secs--))
        done
}

countdown 10 "doing stuff"
```

Reply:
Thanks!  I updated the relevant script to make use of this.